R/c.R

```diff
@@ -39,8 +39,8 @@ vec_c <- function(..., .ptype = NULL) {
     return(NULL)
 
   ns <- map_int(args, vec_size)
-  out <- vec_na(ptype, sum(ns))
-  if (is.null(names(args))) {
+  out <- vec_na(unname(ptype), sum(ns))
```

Comment:
This doesn't feel quite right to me. Should the names be stripped in either `vec_na()` or `vec_ptype()`? Otherwise, you need to explain why `vec_c()` needs special behaviour.

Reply:
Removing this change for now.